recipes/optimize.rb

```diff
@@ -6,6 +6,37 @@
 #
 #
 
-sysctl 'vm.max_map_count' do
-  value 262144
+template '/etc/sysctl.conf' do
+  source 'etc/default/sysctl_conf.erb'
+  owner 'root'
+  group 'root'
+  mode '0644'
+  variables(:max_queued_events => node[cookbook_name][:max_queued_events],
+            :max_user_instances => node[cookbook_name][:max_user_instances],
+            :max_user_watches => node[cookbook_name][:max_user_watches],
+            :max_map_count => node[cookbook_name][:max_map_count],
+            :dmesg_restrict => node[cookbook_name][:dmesg_restrict],
+            :ipv4_gc_thresh3 => node[cookbook_name][:ipv4_gc_thresh3],
+            :ipv6_gc_thresh3 => node[cookbook_name][:ipv6_gc_thresh3],
+            :maxkeys => node[cookbook_name][:maxkeys]
+            )
 end
+
+template '/etc/security/limits.conf' do
+  source 'etc/default/limits.erb'
+  owner 'root'
+  group 'root'
+  mode '0644'
+  variables(:soft_nofile => node[cookbook_name][:soft_nofile],
+            :hard_nofile => node[cookbook_name][:hard_nofile],
+            :root_soft_nofile => node[cookbook_name][:root_soft_nofile],
+            :root_hard_nofile => node[cookbook_name][:root_hard_nofile],
+            :soft_memlock => node[cookbook_name][:soft_memlock],
+            :hard_memlock => node[cookbook_name][:hard_memlock])
+end
+
+reboot 'app_requires_reboot' do
```

Comment:
do we absolutely have to reboot? can we do combination of:
1. changing the file
2. do `sysctl -w`

I'm not sure about `/etc/security/limits.conf` though, can it be done without reboot?
cc: @fadlinurhasan

Reply:
Noted for combination of file changing and sysctl -w

Reading some references, changing `/etc/security/limits.conf` only take effect after rebooting

Reply:
Given the circumstances, should we proceed with reboot process in the end of client chef run ?